### PR TITLE
test: Ensure migrations are run before all tests

### PR DESCRIPTION
**Pull request by lynnagara**
Thi ensures that migrations are run before any tests, not just those
that inherit from the BaseTest class. This should enable us to simplify
the way we write tests and deprecate BaseTest in future.

### CODE REVIEW

File: tests/conftest.py [hunk 1]
```diff
@@ -18,16 +22,14 @@ def pytest_configure() -> None:
     """
     setup_sentry()
 
-    # There is only one cluster in test, so fetch the host from there.
-    cluster = settings.CLUSTERS[0]
+    for cluster in settings.CLUSTERS:
```
Comment (fpacifici) [inline tests/conftest.py:25]: Is there not a single cluster in test anymore ?

File: tests/conftest.py [hunk 2]
```diff
@@ -38,3 +40,23 @@ def clock() -> Iterator[Clock]:
 @pytest.fixture
 def broker(clock: TestingClock) -> Iterator[LocalBroker[TPayload]]:
     yield LocalBroker(MemoryMessageStorage(), clock)
+
+
+@pytest.fixture(autouse=True)
+def run_migrations() -> Iterator[None]:
```
Comment (fpacifici) [inline tests/conftest.py:46]: Who is using this now ? This logic was in dataset_manager() before, but I don't see it being executed in the tests that were relying on dataset_manager.

File: tests/conftest.py (hunk 1 above)
Comment (lynnagara) [reply tests/conftest.py]: There is still, but it's unnecessary to make that assumption. If we change that in the future, I wanted this to still work.

File: tests/conftest.py (hunk 2 above)
Comment (lynnagara) [reply tests/conftest.py:41]: It's run before every test now regardless of whether they use dataset_manager (because of `autouse=True`)

Comment (fpacifici) [reply tests/conftest.py:41]: ah ok, nice

**Review by fpacifici — APPROVED** [review]

**Pull request merged**